poi/src/main/java/org/apache/poi/hssf/model/InternalWorkbook.java

```diff
@@ -1824,175 +1811,32 @@ public PaletteRecord getCustomPalette() {
-            dgg.setDrawingsSaved(0);
-            dgg.setFileIdClusters(new EscherDggRecord.FileIdCluster[] {} );
-            drawingManager = new DrawingManager2(dgg);
-            EscherContainerRecord bstoreContainer = null;
-            if (!escherBSERecords.isEmpty())
-            {
-                bstoreContainer = new EscherContainerRecord();
-                bstoreContainer.setRecordId( EscherContainerRecord.BSTORE_CONTAINER );
-                bstoreContainer.setOptions( (short) ( (escherBSERecords.size() << 4) | 0xF ) );
-                for (EscherRecord escherRecord : escherBSERecords) {
-                    bstoreContainer.addChildRecord( escherRecord );
-                }
-            }
-            opt.setRecordId((short) 0xF00B);
-            opt.setOptions((short) 0x0033);
-            opt.addEscherProperty( new EscherBoolProperty(EscherPropertyTypes.TEXT__SIZE_TEXT_TO_FIT_SHAPE, 524296) );
-            opt.addEscherProperty( new EscherRGBProperty(EscherPropertyTypes.FILL__FILLCOLOR, 0x08000041) );
-            opt.addEscherProperty( new EscherRGBProperty(EscherPropertyTypes.LINESTYLE__COLOR, 134217792) );
-            splitMenuColors.setRecordId((short) 0xF11E);
-            splitMenuColors.setOptions((short) 0x0040);
-            splitMenuColors.setColor1(0x0800000D);
-            splitMenuColors.setColor2(0x0800000C);
-            splitMenuColors.setColor3(0x08000017);
-            splitMenuColors.setColor4(0x100000F7);
-
-            dggContainer.addChildRecord(dgg);
-            if (bstoreContainer != null) {
-                dggContainer.addChildRecord( bstoreContainer );
-            }
-            dggContainer.addChildRecord(opt);
-            dggContainer.addChildRecord(splitMenuColors);
-
-            int dgLoc = findFirstRecordLocBySid(DrawingGroupRecord.sid);
-            if (dgLoc == -1) {
-                DrawingGroupRecord drawingGroup = new DrawingGroupRecord();
-                drawingGroup.addEscherRecord(dggContainer);
-                int loc = findFirstRecordLocBySid(CountryRecord.sid);
-
-                getRecords().add(loc+1, drawingGroup);
-            } else {
-                DrawingGroupRecord drawingGroup = new DrawingGroupRecord();
-                drawingGroup.addEscherRecord(dggContainer);
-                getRecords().set(dgLoc, drawingGroup);
-            }
-
-        }
-    }
-
     public WindowOneRecord getWindowOne() {
         return windowOne;
     }
 
+    /**
+     * Returns the BSE record at the given index.
+     *
+     * @param pictureIndex 1-based index of the picture for which to retrieve the BSE record.
+     * @return BSE record for the picture at the 1-based index.
+     * @throws IndexOutOfBoundsException if the index is larger than the number of pictures available.
+     */
     public EscherBSERecord getBSERecord(int pictureIndex) {
-        return escherBSERecords.get(pictureIndex-1);
+        return getDrawingManager().getBSERecord(pictureIndex);
```

Comment:
okay i basically see that we moved the drawingManager code to DrawingManager2 and made calls to getDrawingManager() here, why are we doing this?

also is getting rid of the `pictureIndex-1` call here intended as well?

Reply:
does getDrawingManager() being called this way add any performance costs over the original implementation?